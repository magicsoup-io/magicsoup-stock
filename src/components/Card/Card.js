import styled from 'styled-components'
import {
  borders,
  borderColor,
  borderRadius,
  boxShadow,
  backgroundImage,
  backgroundSize,
  backgroundPosition,
  backgroundRepeat,
  opacity,
  variant,
} from 'styled-system'

import Box from '../Box'

const themed = key => props => props.theme[key]

const cards = variant({ key: 'cards' })

const Card = styled(Box)`
  ${borders}
  ${borderColor}
  ${borderRadius}
  ${boxShadow}
  ${backgroundImage}
  ${backgroundSize}
  ${backgroundPosition}
  ${backgroundRepeat}
  ${opacity}
  ${cards}
  ${themed('Card')}
`
/** @component */
export default Card

Card.propTypes = {
  ...borders.propTypes,
  ...borderColor.propTypes,
  ...borderRadius.propTypes,
  ...boxShadow.propTypes,
  ...backgroundImage.propTypes,
  ...backgroundSize.propTypes,
  ...backgroundPosition.propTypes,
  ...backgroundRepeat.propTypes,
  ...opacity.propTypes,
  ...cards.propTypes,
}
