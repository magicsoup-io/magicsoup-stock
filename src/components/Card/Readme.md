Card:

```jsx
<Card
  fontSize={3}
  fontWeight='bold'
  width={[ 1, 1, 1/2 ]}
  p={5}
  my={5}
  bg='secondary'
  borderRadius={8}
  color='white'
  boxShadow='0 2px 16px rgba(0, 0, 0, 0.25)'
>
  Card
</Card>
```
