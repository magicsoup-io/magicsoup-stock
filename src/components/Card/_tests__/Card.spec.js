import React from 'react'
import renderer from 'react-test-renderer'

import Card from '../Card'

it('renders <Card /> without throwing', () => {
  const tree = renderer.create(<Card> I am a Card</Card>).toJSON()
  expect(tree).toMatchSnapshot()
})
