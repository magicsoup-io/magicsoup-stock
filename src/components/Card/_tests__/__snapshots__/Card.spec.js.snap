// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Card /> without throwing 1`] = `
<div
  className="sc-bdVaJa sc-bwzfXH dIZceY"
>
   I am a Card
</div>
`;
